src/bin/report_collector.rs

```diff
@@ -255,46 +257,53 @@ async fn ipa(
         r
     };
 
-    // `key_registries` holds the owned registries. `match_key_encryption` has the key ID and
-    // registry references that we pass to playbook_ipa.
     let mut key_registries = KeyRegistries::default();
-    let match_key_encryption = if ipa_query_config.plaintext_match_keys {
-        None
-    } else {
-        match key_registries.init_from(network) {
-            mk_enc @ Some(_) => mk_enc,
-            None => panic!(
-                "match key encryption was requested, but one or more helpers is missing a public key"
-            ),
-        }
-    };
-
     let actual = playbook_ipa::<Fp32BitPrime, MatchKey, BreakdownKey, _>(
         &input_rows,
         &helper_clients,
         query_id,
-        match_key_encryption,
+        ipa_query_config,
+        key_registries.init_from(network),
     )
     .await;
 
     tracing::info!("{m:?}", m = ipa_query_config);
 
-    validate(&expected, &actual);
+    validate(&expected, &actual.breakdowns);
 
     if let Some(ref path) = args.output_file {
-        let file = File::options()
+        // it will be sad to lose the results if file already exists.
+        let path = if Path::is_file(&path) {
+            let mut new_file_name = thread_rng()
+                .sample_iter(&Alphanumeric)
+                .take(5)
+                .map(char::from)
+                .collect::<String>();
+            let file_name = path.file_stem().ok_or("not a file")?;
+
+            new_file_name.insert(0, '-');
+            new_file_name.insert_str(0, &file_name.to_string_lossy());
+            tracing::warn!(
+                "{} file exists, renaming to {:?}",
+                path.display(),
+                new_file_name
+            );
+
+            // it will not be 100% accurate until file_prefix API is stabilized
+            Cow::Owned(
+                path.with_file_name(&new_file_name)
+                    .with_extension(path.extension().unwrap_or("".as_ref())),
+            )
+        } else {
+            Cow::Borrowed(path)
+        };
+        let mut file = File::options()
             .write(true)
             .create_new(true)
-            .open(path)
+            .open(path.deref())
             .map_err(|e| format!("Failed to create output file {}: {e}", path.display()))?;
 
-        let mut writer = BufWriter::new(file);
-        for (i, row) in actual.iter().enumerate() {
-            if i > 0 {
-                write!(writer, ",")?;
-            }
-            write!(writer, "{}", row)?;
-        }
+        write!(file, "{}", serde_json::to_string_pretty(&actual)?)?;
```

Comment:
Lets also add a print with output file name. 
"The results of query were written in {path}"

Reply:
yea I thought it may be redundant because caller specifies the path via `--output-file` argument